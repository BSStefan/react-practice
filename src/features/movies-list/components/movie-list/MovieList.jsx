// @flow
import * as React from 'react';
import { Row, Col } from 'react-bootstrap';
import MovieListItem from '../movie-list-item/MovieListItem';

import type { MovieListResponse } from '../../store/types'

type Props = {
  movieList: MovieListResponse,
}

const MovieList = (props: Props) => {
  const { movieList } = props;
  return (
    <Row>
      {movieList.movies.map(movie => (
        <Col lg={3} sm={6} xs={12} key={movie.id}>
          <MovieListItem movie={movie} />
        </Col>
      ))}
    </Row>
  )
};

export default MovieList;