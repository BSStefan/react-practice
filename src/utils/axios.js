// @flow
import axios from 'axios';

export const setTmdbKey = (key: string): void => {
  axios.defaults.params = {};
  axios.defaults.params['api_key'] = key;
};

export const setBaseUrl = (url: string): void => {
  axios.defaults.baseURL = url;
}