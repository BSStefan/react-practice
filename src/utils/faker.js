// @flow
import type { Movie } from '../types/Movie';

export const fakeMovie = ({ id, title, original_language, genre_ids, overview, poster, release_date }: Movie = {}): Movie => ({
  id: id || 123,
  title: title || 'Movie title',
  original_language: original_language || 'en',
  genre_ids: genre_ids || ['123'],
  overview: overview || 'Test test',
  poster: poster || 'http://test.com',
  release_date: release_date || '12/12/2019',
});
