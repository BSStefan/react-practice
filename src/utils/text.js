// @flow

export const shortText = (text: string, maxLenght: number): string => text.length > maxLenght ? `${text.slice(0, maxLenght - 5)} ...` : text;