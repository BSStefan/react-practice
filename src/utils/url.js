export const TMDB_URL = 'https://api.themoviedb.org/3';
export const TMDB_POSTER_URL = 'https://image.tmdb.org/t/p/w185_and_h278_bestv2'