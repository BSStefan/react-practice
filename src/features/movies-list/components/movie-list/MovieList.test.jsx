import * as React from 'react';
import { shallow } from 'enzyme';
import MovieList from './MovieList';
import { fakeMovie } from '../../../../utils/faker';
import * as renderer from 'react-test-renderer';

const defaultProps = {
  movieList: {
    page: 1,
    total_results: 10,
    total_pages: 2,
    movies: [fakeMovie()]
  },
}


it('Should render omponent without craching', () => {
  const wrapper = shallow(<MovieList {...defaultProps} />)

  expect(wrapper.exists()).toBe(true);
})

it('Should find one list item', () => {
  const wrapper = shallow(<MovieList {...defaultProps} />)

  const listItems = wrapper.find('MovieListItem')

  expect(listItems.length).toBe(1);
})

it('Should match snapshot', () => {
  const tree = renderer.create(<MovieList {...defaultProps} />).toJSON();

  expect(tree).toMatchSnapshot();
})

