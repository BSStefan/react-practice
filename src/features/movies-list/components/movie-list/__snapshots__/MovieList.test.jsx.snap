// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-3 col-sm-6 col-12"
  >
    <div
      className="sc-bdVaJa hniPbm card"
      id={123}
    >
      <img
        className="card-img-top"
        src="https://image.tmdb.org/t/p/w185_and_h278_bestv2http://test.com"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Movie title
        </div>
        <p
          className="card-text"
        >
          Test test
        </p>
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          View Movie
        </button>
      </div>
    </div>
  </div>
</div>
`;
